src/storage/indexed-db-adapter.js

```diff
@@ -13,6 +15,31 @@ export default class IndexedDBAdapter {
     this.errors = []
   }
 
+  async checkAndUpdate (wordItem, segment, currentRemoteItems) {  
+    if (segment === 'context' || !segment)  {
+      if (currentRemoteItems.length > 0) {
+        wordItem.context = []
+        for(let contextItem of currentRemoteItems[0].context) {
+          wordItem.context.push(WordItem.readContext([contextItem])[0])
+        }
+      }
+    }
+
+    let currentLocalItems = await this.query({ wordItem })
+
+    if (currentLocalItems.length === 0) {
+      await this.update(wordItem, { segment: 'common' })  
+    }
+
+    if (!segment) {
```

Comment:
if segment wasn't supplied, won't this result in the 'common' segment being updated twice? first on line 31 and then again on line 38?